### PR TITLE
:sparkles: add prefix filtering

### PR DESCRIPTION
**Pull request by ezrasilvera**
## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/clustermanager/provider.go [hunk 1]
```diff
@@ -38,11 +40,12 @@ func ProviderNS(name string) string {
...
-	providerClient  clusterprovider.ProviderClient
-	c               *controller
-	providerWatcher clusterprovider.Watcher
-	nameSpace       string
+	name             string
+	providerClient   clusterprovider.ProviderClient
+	c                *controller
+	providerWatcher  clusterprovider.Watcher
+	nameSpace        string
+	discoveryPresfix string
```
Comment (andreyod) [inline pkg/clustermanager/provider.go:48]: Typo `discoveryPresfix` -> `discoveryPrefix`

File: pkg/clustermanager/controller.go [hunk 2]
```diff
@@ -45,6 +45,13 @@ const (
 	controllerName                                   = "logicalcluster-manager"
 )
 
+func LCKeyFunc(ns string, name string) string {
```
Comment (andreyod) [inline pkg/clustermanager/controller.go:48]: Looks like this is a local helper function so it shouldn't be exported and maybe should be moved to provider.go where it is called.

File: pkg/clustermanager/provider.go [hunk 3]
```diff
@@ -75,17 +80,33 @@ func CreateProvider(c *controller, providerName string,
 		return nil, errors.New("unknown provider type")
 	}
 
+	providerDescObj, found, _ := c.clusterProviderInformer.GetIndexer().GetByKey(LCKeyFunc("", providerName))
+	if found {
+		providerDesc, ok := providerDescObj.(*lcv1alpha1apis.ClusterProviderDesc)
+		if ok {
+			discoveryPresfix = providerDesc.Spec.ClusterPrefixForDiscovery
```
Comment (andreyod) [inline pkg/clustermanager/provider.go:87]: What happens if it's an empty string?

File: pkg/clustermanager/provider.go [hunk 4]
```diff
@@ -113,21 +134,34 @@ func (p *provider) StopDiscovery() error {
...
 
+		if found {
+			reflcluster, ok = reflclusterObj.(*lcv1alpha1apis.LogicalCluster)
+			if !ok {
+				runtime.HandleError(errors.New("unexpected object type. expected ClusterProviderDesc"))
+				continue
+			}
+		}
+
+		if !found || (found && !reflcluster.Spec.Managed) {
```
Comment (andreyod) [inline pkg/clustermanager/provider.go:155]: This logic probably deserves a comment.

File: pkg/clustermanager/provider.go (hunk 1 above)
Comment (ezrasilvera) [reply pkg/clustermanager/provider.go:48]: fixed

File: pkg/clustermanager/controller.go (hunk 2 above)
Comment (ezrasilvera) [reply pkg/clustermanager/controller.go:48]: fixed

**Pull request merged**